src/app/components/album-details/album-details.service.js

```diff
@@ -0,0 +1,24 @@
+export class AlbumDetailsService {
+  constructor($http) {
+    'ngInject';
+    this.$http = $http;
+  }
+  
+  getAlbum(id) {
+    return this.$http.get(`/album/${id}`)
+            .then(response => response.data)
+            .catch(error => { throw error; });
```

Comment:
Don't know if make sense to throw the error because it doesn't let you manage the error properly in a real case (prod. ready).

And handling an A+ Promise inside a try-catch is a bad practice because try-catch forces to be synchronous